src/RateLimiter/CounterInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Web\RateLimiter;
+
+/**
+ * CounterInterface describes rate limiter counter
+ */
+interface CounterInterface
+{
+    /**
+     * @param string $id set counter ID
+     * Counters with non-equal IDs are counted separately.
+     */
+    public function setId(string $id): void;
+
+    /**
+     * Counts one request as done and returns result containing statistics and rate limiting decision
+     * @return CounterStatistics
+     */
+    public function incrementAndGetResult(): CounterStatistics;
```

Comment:
This method should be divided into 2 methods: `increment` and `getResult`. 
See [CQRS](wikipedia.org/wiki/CQRS)

Reply:
if divided, atomic operation is not possible.

Reply:
which one?

Reply:
`incrementAndGetResult`

Reply:
That's not the case to apply CQRS to. There may be a valid reason to have two methods as described by https://github.com/yiisoft/yii-web/pull/204#discussion_r366701262 though. Let's discuss there.